L/llama_cpp/bundled/patches/fix-mingw32-windows-include.patch

```diff
@@ -0,0 +1,19 @@
+diff --git a/llama.cpp b/llama.cpp
+index 854bb89..896b95b 100644
+--- a/llama.cpp
++++ b/llama.cpp
+@@ -13,9 +13,14 @@
+ #include <cstring>
+ 
+ #if defined(_WIN32) && !defined(_POSIX_MAPPED_FILES)
++#if !defined(__MINGW32__)
+ #define WIN32_LEAN_AND_MEAN
+ #include <Windows.h>
+ #else
++// ref: https://github.com/ggerganov/whisper.cpp/issues/168
++#include <windows.h>
++#endif
++#else
```

Comment:
No, this should ***always*** be `windows.h`.

Reply:
The difference isn't MinGW vs else, but case-sensitive vs case-insensitive file systems.

Reply:
You mean also when compiling with on non-mingw32 (gcc) compilers on windows, i.e. visual studio c++? I don't really have a means of testing this setup myself (honestly I don't use windows, and it always seems to cause the most work :smile: ).

I copied the logic from https://github.com/ggerganov/llama.cpp/blob/53dbba769537e894ead5c6913ab2fd3a4658b738/ggml.c#L31-L36 and assumed there was a reason for the lower-case / upper-case includes.

Reply:
> You mean also when compiling with non-mingw32 (gcc) compilers on windows, i.e. visual studio c++?

Yes, because as I said the problem isn't the toolchain used but the file system used.

Reply:
> I copied the logic from [ggerganov/llama.cpp@53dbba7/ggml.c#L31-L36](https://github.com/ggerganov/llama.cpp/blob/53dbba769537e894ead5c6913ab2fd3a4658b738/ggml.c#L31-L36) and assumed there was a reason for the lower-case / upper-case includes.

That's a.... stupid logic......

Reply:
Thanks again for the explanation! Sorry I didn't see your comment about the windows case-insensitive filesystem until after I had sent my comment off.

`windows.h` it is.